index.d.ts

```diff
@@ -0,0 +1,93 @@
+import { AxiosInstance } from 'axios';
+
+export interface Client {
+  readonly app: { slug: string };
+
+  triggerBuild(buildParams?: BuildParameters): Promise<Build>;
+}
+
+export interface AbortOptions {
+  readonly reason?: string;
+  readonly withSuccess?: boolean;
+  readonly skipNotifications?: boolean;
+}
+
+export interface FollowOptions {
+  readonly heartbeat?: number;
+  readonly interval?: number;
+}
+
+export interface AbortResponse {
+  readonly status?: string;
+  readonly message?: string;
+}
+
+export interface BuildDescription {
+  readonly abort_reason: string;
+  readonly branch: string;
+  readonly build_number: integer;
+  readonly commit_hash: string;
+  readonly commit_message: string;
+  readonly commit_view_url: string;
+  readonly environment_prepare_finished_at: string;
+  readonly finished_at: string;
+  readonly is_on_hold: boolean;
+  readonly original_build_params: string;
+  readonly pull_request_id: integer;
+  readonly pull_request_target_branch: string;
+  readonly pull_request_view_url: string;
+  readonly slug: string;
+  readonly stack_config_type: string;
+  readonly stack_identifier: string;
+  readonly started_on_worker_at: string;
+  readonly status: integer;
+  readonly status_text: string;
+  readonly tag: string;
+  readonly triggered_at: string;
+  readonly triggered_by: string;
+  readonly triggered_workflow: string;
+}
+
+export interface Build {
+  abort(
+    options?: Pick<AbortOptions, Exclude<AbortOptions, 'reason'>>
+  ): Promise<void>;
+  abort(options?: AbortOptions): Promise<AbortResponse>;
+
+  describe(): Promise<BuildDescription>;
+
+  follow(options?: FollowOptions): Promise<void>;
+
+  isFinished(): Promise<Boolean>;
+}
+
+export interface CommitPathsFilter {
+  readonly added?: string[];
+  readonly modified?: string[];
+  readonly removed?: string[];
+}
+
+export interface BuildOptions {
```

Comment:
Is it really true that every field is optional? I would have expected something to be required

Reply:
Technically, the schema has no `required` properties. But the docs say:
> Specify an app slug and at least one parameter out of three: a git tag or git commit hash, a branch, or a workflow ID.

See: https://api-docs.bitrise.io/#/builds/build-trigger.

Perhaps I could add a type overload to extend from the base parameters and have at least one of those specified properties present. Or maybe a comment is sufficient?